src/BullOak.Repositories.EventStore/EventstoreRepository.cs

```diff
@@ -16,12 +17,21 @@ public EventStoreRepository(IHoldAllConfiguration configs, IEventStoreConnection
             this.connection = connection ?? throw new ArgumentNullException(nameof(connection));
         }
 
-        public async Task<IManageAndSaveSessionWithSnapshot<TState>> BeginSessionFor(TId id, bool throwIfNotExists = false)
+        public async Task<IManageSessionOf<TState>> BeginSessionFor(TId id, bool throwIfNotExists = false)
         {
             var session = new EventStoreSession<TState>(configs, connection, id.ToString());
             await session.Initialize();
             return session;
         }
 
+        public Task<bool> Contains(TId selector)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task Delete(TId selector)
+        {
+            throw new NotImplementedException();
```

Comment:
Just checking if it is OK to use generic `NotImplementedException` or it is worth using a more BullOak-specific one?

Reply:
This is wrong, but I'll fix it in another PR. I maybe should have added a comment to say so though.